src/CMakeLists.txt

```diff
@@ -45,7 +45,7 @@ add_subdirectory(util)
 add_subdirectory(plugin)
 add_subdirectory(core)
 
-add_executable(linuxdeploy main.cpp)
+add_executable(linuxdeploy main.cpp linuxdeploy.cpp)
```

Comment:
What is that `linuxdeploy.cpp` file supposed to do? Is it a utility module? Does it encapsulate specific functionality? I have no clue. The name doesn't help me at all. Please think of a better one.

Reply:
I know, I guess that it should encapsulate the `linuxdeploy` functionality. I like to partition away main, form app logic from core (shared) components. 

The rationale behind this partitioning scheme:
- core (shared) components can be used in other projects so the must not depend on linuxdeploy specific stuff.
- linuxdeploy should be configurable and only describe the major workflows
- main do configure linuxdeploy and put all the partitions together.

Reply:
It's not exactly cli related... it's just part of the workflow
In the future, I would like to move the other parts of the workflow into functions as well and add a unit test to them too.

Reply:
But it's used in the CLI/frontend exclusively. liblinuxdeploy doesn't define the workflow, the frontend does.

Reply:
the front end is about parsing CLI arguments this has nothing to do with that

Reply:
No, the frontend is the linuxdeploy program *as a whole* from my point of view. The backend is liblinuxdeploy.

Reply:
Let's use core.h and core.cpp in the meanwhile